handler.go

```diff
@@ -168,14 +167,36 @@ func (s *RPCServer) handleReader(ctx context.Context, r io.Reader, w io.Writer,
 				s.maxRequestSize))
 		return
 	}
+	// check if bufferedRequest is array or not.
+	if bufferedRequest.Bytes()[0] != '[' {
```

Comment:
Are we sure there's no way to send a request with an empty body? We probably need to check if the request is empty before continuing.